src/core/frames/frame_renderer.ts

```diff
@@ -29,6 +29,7 @@ export class FrameRenderer extends Renderer<FrameElement> {
     if (sourceRange) {
       sourceRange.selectNodeContents(frameElement)
       this.currentElement.appendChild(sourceRange.extractContents())
+      frameElement.disconnectedCallback()
```

Comment:
I'm not sure of a way to add a test to guard against a regression.

The [getEventListeners()](https://developer.chrome.com/docs/devtools/console/utilities/#getEventListeners-function) function used to uncover this leak is defined in the console's suite of tools, and isn't available to the document.

I'm open to ideas!

Reply:
Maybe attach a listener that increments a global you can check?

Reply:
I don't think that the code within the `connectedCallback()` and `disconnectedCallback()` fire any events directly. It's also worth noting that we're manually invoking these methods within Turbo itself, so they're kind of "synthetic" connections, which makes a `MutationObserver` similarly unpredictable, since the underlying element itself might not be attached to the browser's document.